src/controllers/AllBooksController.js

```diff
@@ -48,24 +56,36 @@ const getAllBooks = async (req, res) => {
         const fieldsList = fields.split(',').join(' ')
         result = result.select(fieldsList)
     }
+    if (longitude && latitude) {
+        const closestUserAddress = await findClosestUserAddress(
+            latitude,
+            longitude,
+            searchRadius
+        )
+
+        result = closestUserAddress
+    }
```

Comment:
Up above you've already retrieved items based on existing queries. I would recommend passing that list of books into findClosestUserAddress and filtering through those based on location.

Reply:
If you pass down the list of books, since you're populating with long/lat you shouldn't even need to query the users in findClosestUserAddress.